nucypher/blockchain/eth/clients.py

```diff
@@ -189,7 +189,12 @@ def inject_middleware(self, middleware, **kwargs):
 
     @property
     def chain_id(self) -> int:
-        return int(self.w3.eth.chainId, 16)
+        try:
+            # from hex-str
+            return int(self.w3.eth.chainId, 16)
+        except TypeError:
```

Comment:
:+1:

Reply:
@KPrasch do we need to be careful here? - in newer versions chainId could already be in decimal: See https://github.com/ethereum/web3.py/pull/1394. See release note for v5.0.0

Reply:
I think the intention is to be backwards-compatible

Reply:
Ah right!

Reply:
:heavy_check_mark: